auth/authenticators/iam-authenticator.ts

```diff
@@ -14,17 +14,38 @@
  * limitations under the License.
  */
 
-import { OutgoingHttpHeaders } from 'http';
 import { IamTokenManager } from '../token-managers';
 import { validateInput } from '../utils';
-import { BaseOptions, TokenRequestBasedAuthenticator } from './token-request-based-authenticator';
+import { BaseOptions, TokenRequestBasedAuthenticator }
+  from './token-request-based-authenticator';
 
+/** Configuration options for IAM authentication. */
 export interface Options extends BaseOptions {
+  /** The IAM api key */
   apikey: string;
+  /**
+   * The `clientId` and `clientSecret` fields are used to form a "basic"
+   * authorization header for IAM token requests.
+   */
   clientId?: string;
+  /**
+   * The client_id and client_secret fields are used to form a "basic"
```

Comment:
```suggestion
   * The `clientId` and `clientSecret` fields are used to form a "basic"
```

I only commented on one instance of this but there were a few